src/platform_impl/linux/x11/xdisplay.rs

```diff
@@ -91,6 +91,14 @@ impl XConnection {
             conn.map_err(|e| XNotSupported::XcbConversionError(Arc::new(WrapConnectError(e))))?
         };
 
+        // Make sure Xlib knows XCB is handling events.
+        unsafe {
+            (xlib_xcb.XSetEventQueueOwner)(
+                display,
+                x11_dl::xlib_xcb::XEventQueueOwner::XCBOwnsEventQueue,
+            );
+        }
```

Comment:
Since the commit message for this one says

> I don't know what this actually does, but it can't hurt, right?

"Normally", all events and errors go through Xlib. It does some gymnastics internally to ensure that before the reply for a request is handled, everything that happens before is done: Events are in its internal event queue and errors where passed to the error handler. This has the effect of removing all events/errors from XCB's queue.

Thus, normally you cannot use `xcb_poll/wait_for_event()` when Xlib is involved, since Xlib might already have "eaten" an event and put it into its own queue.

The call you are doing here just tells Xlib "do not mess with the event queue, kthxbye". It still will use the flag `XCB_REQUEST_CHECKED` flags when sending requests to that any errors caused "through" Xlib are reported to Xlib via `xcb_wait_for_reply()` and do not end up in XCB's event queue, but it will no longer call `xcb_poll_for_event()` or `xcb_wait_for_event()`.